src/cdc/runner.zig

```diff
@@ -1435,3 +1440,54 @@ test "amqp: JSON message" {
         ).diff(buffer);
     }
 }
+
+test "amqp: metrics" {
+    var summary: Metrics.TimingSummary = .{
+        .timer = try std.time.Timer.start(),
+    };
+
+    try testing.expect(summary.count == 0);
+    try testing.expect(summary.event_count == 0);
+    try testing.expect(summary.duration_sum.ns == 0);
+    try testing.expect(summary.duration_max == null);
+    try testing.expect(summary.duration_min == null);
+
+    summary.record(10);
+
+    try testing.expect(summary.count == 1);
+    try testing.expect(summary.event_count == 10);
+    try testing.expect(summary.duration_sum.ns >= 0);
+    try testing.expect(summary.duration_min != null);
+    try testing.expect(summary.duration_max != null);
+    try testing.expect(summary.duration_sum.ns == summary.duration_min.?.ns);
+    try testing.expect(summary.duration_sum.ns == summary.duration_max.?.ns);
+    try testing.expect(summary.duration_min.?.ns == summary.duration_max.?.ns);
+
+    std.time.sleep(100 * std.time.ns_per_us);
```

Comment:
This is quite suboptimal, as that more or less adds 100 micros to our overall runtime, which will add up over time. As a rule of thumb, tests should never ever sleep (being rule of thumb, it applies in strictly less than 100% cases). 

We generally prefer using fake time in tests: it runs faster, and you could also test tricky cases like time not advancing at all. 

See, eg, `test "timing overflow"` for en example of this kind of test. 

I don't feel super strongly here, and I want to get the bug fix in the next release, but please consider if you might want to switch to deterministic time here!